include/git2/remote.h

```diff
@@ -488,6 +495,22 @@ typedef int GIT_CALLBACK(git_push_update_reference_cb)(const char *refname, cons
  */
 typedef int GIT_CALLBACK(git_url_resolve_cb)(git_buf *url_resolved, const char *url, int direction, void *payload);
 
+/**
+ * Callback to inform application about which events to wait for
+ *
+ * This callback is used while an asynchroneous network operation is in
+ * progress. It informs the application which events to wait for on a
+ * file descriptor, and how long that timeout should be.
+ *
+ * @param fd The socket to wait for
+ * @param event Event flags
+ * @param timeout Timeout value in seconds
+ * @param payload Payload provided by application
+ * @return GIT_OK on success, any other error code cancels the operation and is passed through to the application
+ */
+
+typedef int GIT_CALLBACK(git_set_fd_events_cb)(git_socket fd, git_event_t event, unsigned int timeout, void *payload);
```

Comment:
Names in libgit2 are always prefixed by `git_`, then the name of the subsystem.  I like to think of it like an object system - so `git_subsystem_action` - eg `git_merge_init` will init a merge, `git_merge_free` will free a merge, etc.

So when I read `git_set_fd_events_cb`, I think that you're `fd_events_cb` on a `git_set`.

I'd suggest `git_remote_set_fd_events_cb` or - terser - `git_remote_events_cb`.

Reply:
yes, fair enough. I will change this in next iteration.